tests-semanticdb/src/test/resources/example/SingleAbstractMethod.scala

```diff
@@ -0,0 +1,16 @@
+package example
+
+class SingleAbstractMethod/*<=example.SingleAbstractMethod#*/ {
+
+  trait SAM/*<=example.SingleAbstractMethod#SAM#*/ {
+    def foo/*<=example.SingleAbstractMethod#SAM#foo().*/(a/*<=example.SingleAbstractMethod#SAM#foo().(a)*/: Int/*=>scala.Int#*/): Int/*=>scala.Int#*/
+  }
+  def withSAM/*<=example.SingleAbstractMethod#withSAM().*/(a/*<=example.SingleAbstractMethod#withSAM().(a)*/: SAM/*=>example.SingleAbstractMethod#SAM#*/) = a/*=>example.SingleAbstractMethod#withSAM().(a)*/.foo/*=>example.SingleAbstractMethod#SAM#foo().*/(0)
+
+  withSAM/*=>example.SingleAbstractMethod#withSAM().*/(_ + 1/*<=example.SingleAbstractMethod#SAM#*/)
+  withSAM/*=>example.SingleAbstractMethod#withSAM().*/((x: Int) => x - 1/*<=example.SingleAbstractMethod#SAM#*/)
```

Comment:
```suggestion
  withSAM/*=>example.SingleAbstractMethod#withSAM().*/((x: Int) => x - 1/*<=local0*/)
```
Otherwise we will find SAM type here, while it's not occurring here and it's just the type of the expression.

For annonymous classes we do locals also.

We would need to make sure isSemanticdbLocal method treats this as a local (since we can never really access this instance from the outside or anywhere for that metter )

Reply:
> first, since withSAM expects a SAM, isn't this a definition of a SAM type? if somebody was looking for places where SAM was instantiated, isn't this one of them? and if it's not, then i might not have fully understood the point of the original problem

This is essentially both a definition on a new SAM type and of an instance of that type. We want occurrence of that instance and type definition in symbols section.

> second, about anonymous classes. is that a separate point or related to the first?

I just mentioned it since it;s a very similar case and we do this already for them. 

> and regarding isSemanticdbLocal: what is that used for?

That's in SymbolOps class, the method is used to know which instances are local to a scope, so no need to name them properly (they are not accessible from the outside scope.) Though this might be more complex than just handling the case in that new method.

Reply:
all addressed in the new patch, please check.